src/vs/workbench/contrib/dialogs/browser/dialog.contribution.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IDialogHandler, IDialogResult, IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { IWorkbenchContribution, IWorkbenchContributionsRegistry, Extensions as WorkbenchExtensions } from 'vs/workbench/common/contributions';
+import { IDialogsModel, IDialogViewItem } from 'vs/workbench/common/dialogs';
+import { HTMLDialogHandler } from 'vs/workbench/contrib/dialogs/browser/dialogHandler';
+import { DialogService } from 'vs/workbench/services/dialogs/common/dialogService';
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+
+export class DialogHandlerContribution implements IWorkbenchContribution {
+	private impl: IDialogHandler;
+
+	private model: IDialogsModel;
+	private currentDialog: IDialogViewItem | undefined;
+
+	constructor(
+		@IDialogService private dialogService: IDialogService,
+		@ILogService logService: ILogService,
+		@ILayoutService layoutService: ILayoutService,
+		@IThemeService themeService: IThemeService,
+		@IKeybindingService keybindingService: IKeybindingService,
+		@IProductService productService: IProductService,
+		@IClipboardService clipboardService: IClipboardService
+	) {
+		this.impl = new HTMLDialogHandler(logService, layoutService, themeService, keybindingService, productService, clipboardService);
+
+		this.model = (this.dialogService as DialogService).model;
```

Comment:
This cast is the consequence of having the handler be in `contrib` and is quite ugly.